GiniVision/Classes/Core/Screens/Help/HelpMenuViewController.swift

```diff
@@ -135,9 +174,24 @@ extension HelpMenuViewController {
     }
     
     override public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if let viewController = viewController(forRowWithId: items[indexPath.row].id) {
-            navigationController?.pushViewController(viewController, animated: true)
+        let item = items[indexPath.row]
+        let viewController = item.viewController
+        viewController.setupNavigationItem(usingResources: backToMenuButtonResource,
+                                           selector: #selector(back),
+                                           position: .left,
+                                           target: self)
+        
+        if let imageNoResultsViewController = viewController as? ImageAnalysisNoResultsViewController {
+            imageNoResultsViewController.didTapBottomButton = {
+                if let cameraViewController = (self.navigationController?
+                    .viewControllers
+                    .compactMap { $0 as? CameraViewController })?
```

Comment:
You know how much I hate iterating view controller in order to find a specific one. It seems like a task for a Coordinator. View controllers shouldn't know about the hierarchy.

Reply:
You are right, and I think we already discuss this in the past. The reason why we didn't use a coordinator here is that the `HelpMenuViewController` is a public screen and that would mean exposing this coordinator.

One thing that we could do to not introduce breaking changes is to add a delegate for those actions that require a coordinator (like navigation), and fallback to the current implementation.

What do you think?

Reply:
That would work for me.